File: PDOO/QytetecJava/src/modeloqytetec/TituloPropiedad.java

package modeloqytetec;

/**
 * @author Jorge Valenzuela Garcia
 */
public class TituloPropiedad {
    
    // ATRIBUTOS
    private String nombre;
    private boolean hipotecada;
    private int alquilerBase;
    private float factorRevalorizacion;
    private int hipotecaBase;
    private int precioEdificar;

    // ATRIBUTOS REFERENCIADOS
    private Jugador propietario;
    private Casilla casilla;
    
    // CONSTRUCTOR
    public TituloPropiedad(String nombre, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar){
        this.nombre = nombre;
        this.hipotecada = false;
        this.alquilerBase = alquilerBase;
        this.factorRevalorizacion = factorRevalorizacion;
        this.hipotecaBase = hipotecaBase;
        this.precioEdificar = precioEdificar;
    }

    // MÉTODOS
    
    protected void cobrarAlquiler(int coste){
        this.propietario.modificarSaldo(coste);
    }
    
    protected int getAlquilerBase(){
        return this.alquilerBase;
    }
    
    protected Casilla getCasilla(){
        return this.casilla;
    }
    
    protected float getFactorRevalorizacion(){
        return this.factorRevalorizacion;
    } 
    
    protected int getHipotecaBase(){
        return this.hipotecaBase;
    }
    
    protected boolean getHipotecada(){
        return this.hipotecada;
    }
   
    protected String getNombre(){
        return this.nombre;
    }
    
    protected int getPrecioEdificar(){
        return this.precioEdificar;
    }
    
    protected boolean propietarioEncarcelado(){
        return this.propietario.getEncarcelado();
    }
    
    protected void setCasilla(Casilla casilla){
        if (casilla.getTipo() == TipoCasilla.CALLE)
            this.casilla = casilla;
    }
    
    protected void setHipotecada(boolean hipotecada){
        this.hipotecada = hipotecada;
    }
    
    protected void setPropietario(Jugador propietario){
        this.propietario = propietario;
    }
    
    protected boolean tengoPropietario(){
        if(this.propietario == null)
            return false;
        else
            return true;
    }
    
    @Override
    public String toString(){
        String salida;
        
        salida= "Nombre: " + this.nombre + "\n" +
                "Hipotecada: " + Boolean.toString(this.hipotecada) + "\n" +
                "Alquiler Base: " + Integer.toString(this.alquilerBase) + "\n" +
                "Factor Revalorizacion: " + Float.toString(this.factorRevalorizacion) + "\n" +
                "Hipoteca Base: " + Integer.toString(this.hipotecaBase)  + "\n" +
                "Precio Edificar: " + Integer.toString(this.precioEdificar) + "\n";
        
                if(this.casilla == null)
                    salida += "Casilla: Sin asignar\n";
                else
                    salida += "Casilla: " + this.casilla.toStringNumeroCasilla() + "\n";
                if(this.propietario == null)
                    salida += "Propietario: Sin asignar\n";
                else
                    salida +="Propietario: " + this.propietario.toStringNombre() + "\n";
        return salida;
    }
    
    public String toStringNombre(){
        return this.nombre;
    }
    
}

File: PDOO/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/titulo_propiedad.rb

module ModeloQytetet
  class TituloPropiedad
    
    def initialize(nombre, alquilerBase, factorRevalorizacion, hipotecaBase, precioEdificar)
      @nombre = nombre
      @hipotecada = false
      @alquiler_base = alquilerBase
      @factor_revalorizacion = factorRevalorizacion
      @hipoteca_base = hipotecaBase
      @precio_edificar = precioEdificar
      @propietario
      @casilla
    end
      
    attr_reader :alquiler_base, :factor_revalorizacion, :hipoteca_base, :nombre, :precio_edificar
    attr_accessor :hipotecada, :casilla
    attr_writer :propietario
    
    
    def to_s
      puts "Nombre #{@nombre} \n Hipotecada: #{@hipotecada} \n AlquilerBase #{@alquiler_base} \n FactorRevalorizacion #{@factor_revalorizacion} \n HipotecaBase: #{@hipoteca_base} \n PrecioEdificar #{@precio_edificar}"
      
      # Comprobamos que exista una casilla
      if @casilla.nil?
        puts "Casilla: Sin asignar"
      else
        puts "Casilla: #{@casilla.num_casilla}"
      end
      
      # Comprobamos que exista un propietario
      if @propietario.nil?
        puts "Propietario: Sin asignar"
      else
        puts "Propietario:"
        puts @propietario.to_s_nombre
      end
    end

    
    def cobrar_alquiler(coste)
      @propietario.modificar_saldo(-coste)
    end
    
    def propietario_encarcelado()
      return @propietario.encarcelado
    end
    
    def tengo_propietario()
      if @propitario.nil?
        return false
      else
        return true
      end
     end
    
  end
end
